projects/ad2s1210_iio/app/ad2s1210_iio.c

```diff
@@ -0,0 +1,619 @@
+				250, AD2S1210_MAX_EXCIT);
+	default:
+
+		break;
+	}
+
+	return len;
+}
+
+static int iio_ad2s1210_attr_set(void *device, char *buf, uint32_t len,
+				 const struct iio_ch_info *channel,
+				 intptr_t priv)
+{
+	struct ad2s1210_dev *dev = (struct ad2s1210_dev *) device;
+	uint16_t fexcit;
+	uint8_t hysteresis;
+	int ret;
+
+	switch (priv) {
+	case RAW_ATTR_ID:
+	case SCALE_ATTR_ID:
+	case LABEL_ATTR_ID:
+	case HYSTERESIS_AVAILABLE_ATTR_ID:
+	case FREQ_AVAIL_ATTR_ID:
+		/* All read-only attributes */
+		break;
+	case FREQ_ATTR_ID:
+		ret = sscanf(buf, "%hu", &fexcit);
+		if (ret != 1)
+			return -EINVAL;
+
+		ret = ad2s1210_reinit_excitation_frequency(dev, fexcit);
+		if (ret)
+			return ret;
+		break;
+
+	case HYSTERESIS_ATTR_ID:
+		ret = sscanf(buf, "%hhu", &hysteresis);
+		if (ret != 1)
+			return -EINVAL;
+
+		ret = ad2s1210_set_hysteresis(dev, hysteresis ? true : false);
+		if (ret)
+			return ret;
+
+		break;
+	default:
+		break;
+	}
+
+	return len;
+}
+
+/*!
+ * @brief       Read the debug register value
+ * @param       dev[in, out]- Pointer to IIO device instance
+ * @param       reg[in]- Register address to read from
+ * @param       readval[out]- Pointer to variable to read data into
+ * @return      0 in case of success, negative value otherwise
+ */
+static int32_t iio_ad2s1210_debug_reg_read(void *dev, uint32_t reg,
+		uint32_t *val)
+{
+	return ad2s1210_reg_read(dev,  (uint8_t) reg, (uint8_t *)val);
+}
+
+/*!
+ * @brief       Write the debug register value
+ * @param       dev[in, out]- Pointer to IIO device instance
+ * @param       reg[in]- Register address to read from
+ * @param       writeval[out]- Pointer to variable to write data into
+ * @return      0 in case of success, negative value otherwise
+ */
+static int32_t iio_ad2s1210_debug_reg_write(void *dev, uint32_t reg,
+		uint32_t val)
+{
+	return ad2s1210_reg_write(dev, (uint8_t)reg, (uint8_t)val);
+}
+
+/**
+ * @brief	Read buffer data corresponding to AD2S1210 IIO device
+ * @param	iio_dev_data[in] - Pointer to IIO device data structure
+ * @return	0 in case of success, negative error code otherwise
+ */
+static int32_t iio_ad2s1210_submit_buffer(struct iio_device_data *iio_dev_data)
+{
+#if (DATA_CAPTURE_MODE == BURST_DATA_CAPTURE)
+	enum ad2s1210_channel chn;
+	uint32_t sample_indx = 0;
+	uint32_t nb_of_samples;
+	uint16_t data;
+	int32_t ret;
+	uint8_t i;
+
+	nb_of_samples = iio_dev_data->buffer->size / BYTES_PER_SAMPLE;
+
+	if (!buf_size_updated) {
+		iio_dev_data->buffer->buf->size = iio_dev_data->buffer->size;
+		buf_size_updated = true;
+	}
+
+	while (sample_indx < nb_of_samples) {
+		/* Push into circular buffer */
+		for (i = 0; i < RESOLVER_CHANNELS; i++) {
+			if (ad2s1210_iio_channels[i].ch_type == IIO_ANGL)
+				chn = AD2S1210_POS;
+			else
+				chn = AD2S1210_VEL;
+
+			if (!(iio_dev_data->buffer->active_mask & (1 << i)))
+				continue;
+
+			ret = ad2s1210_spi_single_conversion(ad2s1210_dev_inst,
```

Comment:
How are we making sure that the sample is being read only after the End of conversion signal?
Continuous mode makes sure that the samples are read only after the falling edge in the PWM pin is detected. How is it done here for burst mode?

Reply:
the no-os driver will toggle the sample pin before a read.

Reply:
So what is the difference between the GPIO_SAMPLE and the external PWM that we are triggering manually?
I am trying to understand how is the end of conversion being handled in continuous vs burst mode. 

Looks like ad2s1210_spi_single_conversion() is being used in  iio_ad2s1210_submit_buffer() and in the ad2s1210_trigger_handler(). So the functionality for reading the sample is the same. But how is end of conversion detected in burst mode?

Reply:
in the case of continues capture  pwm triggers the irq handler that will call ad2s1210_spi_single_conversion where the gpio is toggled before reading from the spi. IOW, the pwm does not directly toggle the sample pin. 

i the case of burst capture,there is no pwm, we call ad2s1210_spi_single_conversion in a loop the number of times that is needed to fill the buffer . each call, the sample gpio is toggled in the ad2s1210_spi_single_conversion function. when the gpio is toggled the sample is captured. after a small delay (350ns) of the toggle the sample can be read. right now i dont specifficaly add a delay in the no-os driver. but i should, and will do.

Reply:
okay. Let us not add the extra delay